pyrsistent/typing.pyi

```diff
@@ -190,25 +188,100 @@ class PList(Sequence[T], Hashable):
     def split(self, index: int) -> Tuple[PList[T], PList[T]]: ...
 
 
-# Types for Checked collections are actually determined by the __type__ dunders,
-# but this should provide better type safety than using Any
+class PClass(Hashable):
+    def __new__(cls, **kwargs: Any): ...
```

Comment:
This can probably use the bounded TypeVar in the same way that PRecord does

Reply:
this should be now better